### PR TITLE
Update proof QR codes

### PR DESCRIPTION
**Pull request by ardier16**

### CODE REVIEW

File: Rarime/Code/Managers/ExternalRequestsManager.swift [hunk 1]
```diff
@@ -9,6 +13,57 @@ class ExternalRequestsManager: ObservableObject {
...
+    func handleRarimeUrl(_ url: URL) {
+        guard let components = URLComponents(url: url, resolvingAgainstBaseURL: false),
+              let params = components.queryItems
+        else {
+            LoggerUtil.common.error("Invalid RariMe app URL: \(url.absoluteString)")
+            AlertManager.shared.emitError(.unknown("Invalid RariMe app URL"))
+            return
+        }
+
+        switch url.host {
```
Comment (IDima1) [inline Rarime/Code/Managers/ExternalRequestsManager.swift:25]: move to let

File: Rarime/Code/Managers/ExternalRequestsManager.swift [hunk 2]
```diff
@@ -9,6 +13,57 @@ class ExternalRequestsManager: ObservableObject {
...
+        case "external":
```
Comment (IDima1) [inline Rarime/Code/Managers/ExternalRequestsManager.swift:26]: move to let

File: Rarime/Code/Managers/ExternalRequestsManager.swift [hunk 3]
```diff
@@ -1,15 +1,74 @@
...
     static let shared = ExternalRequestsManager()
 
-    @Published private(set) var request: ExternalRequest? = nil
+    @Published private(set) var request: ExternalRequests? = nil
+
+    func handleRarimeUrl(_ url: URL) {
+        guard let components = URLComponents(url: url, resolvingAgainstBaseURL: false),
+              let params = components.queryItems
+        else {
+            LoggerUtil.common.error("Invalid RariMe app URL: \(url.absoluteString)")
```
Comment (ivanlele) [inline Rarime/Code/Managers/ExternalRequestsManager.swift:24]: , privacy: public

File: Rarime/Code/Managers/ExternalRequestsManager.swift [hunk 4]
```diff
@@ -1,15 +1,74 @@
...
+            AlertManager.shared.emitError(.unknown("Invalid RariMe app URL"))
+            return
+        }
+
+        switch url.host {
+        case RarimeUrlHosts.external.rawValue:
+            handleExternalRequest(params: params)
+        default:
+            LoggerUtil.common.error("Invalid RariMe URL host: \(url.host ?? "nil")")
```
Comment (ivanlele) [inline Rarime/Code/Managers/ExternalRequestsManager.swift:33]: also

File: Rarime/Code/Managers/ExternalRequestsManager.swift [hunk 5]
```diff
@@ -1,15 +1,74 @@
...
+        }
+    }
+
+    private func handleExternalRequest(params: [URLQueryItem]) {
+        guard let type = params.first(where: { $0.name == "type" })?.value
+        else {
+            LoggerUtil.common.error("Invalid external request URL: \(params)")
```
Comment (ivanlele) [inline Rarime/Code/Managers/ExternalRequestsManager.swift:40]: also

File: Rarime/Code/Managers/ExternalRequestsManager.swift [hunk 6]
```diff
@@ -1,15 +1,74 @@
...
+            AlertManager.shared.emitError(.unknown("Invalid external request URL"))
+            return
+        }
+
+        switch type {
+        case ExternalRequestTypes.proofRequest.rawValue:
+            handleProofRequest(params: params)
+        default:
+            LoggerUtil.common.error("Invalid external request type: \(type)")
```
Comment (ivanlele) [inline Rarime/Code/Managers/ExternalRequestsManager.swift:49]: also

File: Rarime/Code/Managers/ExternalRequestsManager.swift [hunk 7]
```diff
@@ -1,15 +1,74 @@
...
+        }
+    }
+
+    private func handleProofRequest(params: [URLQueryItem]) {
+        guard let rawProofParamsUrl = params.first(where: { $0.name == "proof_params_url" })?.value?.removingPercentEncoding,
+              let proofParamsUrl = URL(string: rawProofParamsUrl)
+        else {
+            LoggerUtil.common.error("Invalid proof request URL: \(params)")
```
Comment (ivanlele) [inline Rarime/Code/Managers/ExternalRequestsManager.swift:57]: also

File: Rarime/Code/Managers/ExternalRequestsManager.swift [hunk 8]
```diff
@@ -1,15 +1,74 @@
...
+            AlertManager.shared.emitError(.unknown("Invalid proof request URL"))
+            return
+        }
+
+        if UserManager.shared.registerZkProof == nil {
+            LoggerUtil.common.error("Proof requests are not available, passport is not registered")
```
Comment (ivanlele) [inline Rarime/Code/Managers/ExternalRequestsManager.swift:63]: also

File: Rarime/Code/Modules/Home/Views/HomeView.swift [hunk 9]
```diff
@@ -286,20 +276,13 @@ struct HomeView: View {
     }
 
     func processQrCode(_ code: String) {
-        let codeData = Data(code.utf8)
-        guard let qrCode = try? JSONDecoder().decode(QrCode.self, from: codeData)
-        else {
+        guard let qrCodeUrl = URL(string: code) else {
+            LoggerUtil.intro.error("Invalid QR code: \(code)")
```
Comment (ivanlele) [inline Rarime/Code/Modules/Home/Views/HomeView.swift:280]: also

**Review by ivanlele — APPROVED** [review]

**Pull request merged**